lib/connection/utils.js

```diff
@@ -78,9 +78,20 @@ var retrieveBSON = function() {
   return BSON;
 }
 
+// Parses the header of a wire protocol message
+var parseHeader = function(message) {
```

Comment:
question: is it worth attempting to check the bounds of the passed in message here? Or is that happening at a higher level? afaict, you just assume there are 16 bytes here which could fail in an unexpected fashion.  If you do end up bounds checking, then please include a unit test for this.